java/api/TypeDBSession.java

```diff
@@ -97,6 +97,18 @@ public interface TypeDBSession extends AutoCloseable {
      */
     void onClose(Runnable function);
 
+    /**
+     * Registers a callback function which will be executed when this session is reopened.
```

Comment:
Let's add a bit more here: when the session is reopened, such as during an automatic failover.